diff --git a/window/src/os/wayland/window.rs b/window/src/os/wayland/window.rs
index 3357bec25..e2bd8dd8b 100644
--- a/window/src/os/wayland/window.rs
+++ b/window/src/os/wayland/window.rs
@@ -907,7 +907,6 @@ impl WaylandWindowInner {
                         ) {
                             self.surface().attach(Some(buffer.wl_buffer()), 0, 0);
                             self.surface().set_buffer_scale(factor as i32);
-                            self.surface().commit();
 
                             self.surface_factor = factor;
                         }

