diff --git a/libavfilter/vf_xpsnr.c b/libavfilter/vf_xpsnr.c
index 3097db0878..c4bf520188 100644
--- a/libavfilter/vf_xpsnr.c
+++ b/libavfilter/vf_xpsnr.c
@@ -237,7 +237,7 @@ static inline double calc_squared_error_and_weight (XPSNRContext const *s,
     }
 
     /* weight += mean squared temporal activity */
-    *ms_act += (double) ta_act / ((double) block_width * (double) block_height);
+    // *ms_act += (double) ta_act / ((double) block_width * (double) block_height);
 
     /* lower limit, accounts for high-pass gain */
     if (*ms_act < (double) (1 << (bit_depth - 6)))
