diff --git a/CMakeLists.txt b/CMakeLists.txt
index de80e98f..0045631e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -403,12 +403,6 @@ endif()
 # binary and symlink
 install(TARGETS Hyprland)
 
-install(
-  CODE "execute_process( \
-        COMMAND ${CMAKE_COMMAND} -E create_symlink \
-        ${CMAKE_INSTALL_FULL_BINDIR}/Hyprland \
-        \"\$ENV{DESTDIR}${CMAKE_INSTALL_FULL_BINDIR}/hyprland\" \
-        )")
 # session file
 install(FILES ${CMAKE_SOURCE_DIR}/example/hyprland.desktop
         DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/wayland-sessions)
